modules/renter/contractor/formcontract.go

```diff
@@ -354,20 +356,27 @@ func (c *Contractor) formContracts(a modules.Allowance) error {
 	endHeight := c.blockHeight + a.Period
 	c.mu.RUnlock()
 	var numContracts uint64
+	var errs []string
 	for _, h := range hosts {
 		_, err := c.newContract(h, filesize, endHeight)
 		if err != nil {
-			// TODO: is there a better way to handle failure here? Should we
-			// prefer an all-or-nothing approach? We can't pick new hosts to
-			// negotiate with because they'll probably be more expensive than
-			// we can afford.
-			c.log.Println("WARN: failed to negotiate contract:", h.NetAddress, err)
+			errs = append(errs, fmt.Sprintf("\t%v: %v", h.NetAddress, err))
 			continue
 		}
 		if numContracts++; numContracts >= a.Hosts {
 			break
 		}
 	}
+	// If we couldn't form any contracts, return an error. Otherwise, just log
+	// the failures.
+	// TODO: is there a better way to handle failure here? Should we prefer an
+	// all-or-nothing approach? We can't pick new hosts to negotiate with
+	// because they'll probably be more expensive than we can afford.
```

Comment:
Uploads should probably be blocked until there are enough contracts to get to a safe redundancy.